### PR TITLE
Add a debug statement for insert rows failure

### PR DESCRIPTION
**Pull request by sfc-gh-japatel**
Found this could be helpful. I wasnt able to figure out why the insertRows API was failing.

### CODE REVIEW

File: src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java [hunk 1]
```diff
@@ -433,6 +433,12 @@ private InsertValidationResponse insertRowsWithFallback(StreamingBuffer buffer)
     Fallback<Object> reopenChannelFallbackExecutorForInsertRows =
         Fallback.builder(() -> insertRowsFallbackSupplier(buffer))
             .handle(SFException.class)
+            .onFailedAttempt(
+                event ->
+                    LOGGER.warn(
+                        String.format(
+                            "Failed Attempt to invoke the insertRows API for buffer:%s", buffer),
```
Comment (sfc-gh-tzhang) [inline src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java:440]: InsertRows shouldn't fail other than invalid or closed channel, what's the reason that you saw?

**Review by sfc-gh-tzhang — APPROVED** [review]

File: src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java (hunk 1 above)
Comment (sfc-gh-japatel) [reply src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java:440]: I did see 429 from one of the APIs. (I think it was blob upload)

Comment (sfc-gh-tzhang) [reply src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java:440]: what do you mean by 429?

Comment (sfc-gh-japatel) [reply src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java:440]: Sorry I meant 429 response code: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429

Comment (sfc-gh-tzhang) [reply src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java:440]: hmmm, that's weird. We're not calling any REST APIs in the InsertRows API, i assume we're talking about `SnowflakeStreamingIngestChannelInternal.insertRows`?

**Pull request merged**